File: taskmanager-web/src/models/DashboardDataDto.ts
export interface JobSummaryDto {
    total: number;
    active: number;
    completed: number;
}

export interface TaskSummaryDto {
    total: number;
    notStarted: number;
    inProgress: number;
    blocked: number;
    completed: number;
}

export interface ProgressDto {
    overallPercent: number;
}

export interface DashboardDataDto {
    jobSummary: JobSummaryDto;
    taskSummary: TaskSummaryDto;
    progress: ProgressDto;
}

File: TaskManager.Application/Features/Dashboard/DTOs/DashboardDataDto.cs
namespace TaskManager.Application.Features.Dashboard.DTOs
{
    public record DashboardDataDto(
         JobSummaryDto JobSummary,
         TaskSummaryDto TaskSummary,
         ProgressDto Progress
     );

    public record JobSummaryDto(
        int Total,
        int Active,
        int Completed
    );

    public record TaskSummaryDto(
        int Total,
        int NotStarted,
        int InProgress,
        int Blocked,
        int Completed
    );

    public record ProgressDto(
        int OverallPercent
    );
}
